### ISSUE TITLE
Add "Import as Maven Project" to the Context menu of resources that contain a pom.xml

### ISSUE THREAD

**Issue by laeubi** [eclipse-m2e/m2e-core#1098]
With [this change](https://github.com/eclipse-tycho/tycho/pull/1740#event-7910105411) is remembered a feature I really miss and I think that greatly degrades user experience:

I have a folder in eclipse that contains a `pom.xml` but there is no easy way to import it as a project (in contrast to one containing a .project file).

# What happen for a folder that **does** contain a `.project` file:
![grafik](https://user-images.githubusercontent.com/1331477/204527554-cb99d6db-b832-4f95-bdc8-1d5ed92ed55a.png)

# What happen for a folder that **does not** contain a `.project` file:
![grafik](https://user-images.githubusercontent.com/1331477/204527989-f446b74f-9d2d-4806-835e-52c1961e3c10.png)

## What needs to be done instead:

1. You click on the "Import..." as shown in above screenshot
2. You need to find and expand "Maven"
3. You have to choose "Existing Maven Project"
4. You have to confirm the single discovered project

## What actually should happen:

There is a menu Item "Import as Maven Project" that simply perform all the steps without any user interaction

**Comment by mickaelistria** [comment]
This issue should be considered more genrally and not only in the scope of m2e. Platform should advertise the "Open Projects from Filesystem" here instead. Then m2e -and others- can do their job without user having to face tons of choices.

**Comment by laeubi** [comment]
> Please don't use any context menu at all. Just hit Help>Perform Setup Tasks. That's why we have the Oomph setups.

This has nothing to do with Oomph setup or not...

> This issue should be considered more genrally and not only in the scope of m2e. Platform should advertise the "Open Projects from Filesystem" here instead.

Choosing Import > Filesystem is even worse as it does not gives me anything to import at all.... even after more than 10 years of using eclipse I have no clue what this dialog is supposed to do or want me to do here
![grafik](https://user-images.githubusercontent.com/1331477/204572130-bc39d702-11b3-4bfa-9f39-204d5e137060.png)

It even has scary options like "overwrite existing resources", no clue what "create top-level folder" should mean.

**Comment by mickaelistria** [comment]
> Choosing Import > Filesystem is even worse as it does not gives me anything to import at all.... even after more than 10 years of using eclipse I have no clue what this dialog is supposed to do or want me to do here

Indeed, this wizard is strange. Please bring this issue to Platform. Maybe it can be deprecated if there is a consensus from most committers that it is not something they use/want to maintain. That would allow to better focus on other more popular and more useful alterantives.

**Comment by laeubi** [comment]
I have now created two followups here:
- https://github.com/eclipse-platform/eclipse.platform.ui/issues/510
- https://github.com/eclipse-platform/eclipse.platform.ui/issues/511
